### PR TITLE
Tool for producing DMR trends

### PR DESCRIPTION
**Pull request by cardinia**
@adewit @connorpa 

#### PR description:

Tool for producing DMR trends for Tracker Alignment:
The README file provides a more detailed description of the tool + some documentation doxygen style has been implemented in the macro.

#### PR validation:

The tool is used for the TkAl offline validation, it should not cause any problem for already existing tools.

### CODE REVIEW

**Comment by pohsun** [comment]
Please put the 4 txt files into `data` directory instead of `interface` directory.

**Comment by fabiocos** [comment]
please test

File: Alignment/OfflineValidation/bin/DMRtrends.cc [hunk 1]
```diff
@@ -0,0 +1,749 @@
...
+		    vector<vector<double>> vectors; 
+		    //if(showlumi&&i<8)scalebylumi(dynamic_cast<TGraph*>(g));
+		    if(showlumi)scalebylumi(g,Year);
+		    g->SetLineColor(*colour);
+                    g->SetMarkerColor(*colour);
+		    TH1F *h = ConvertToHist(g); 
+ 		    h->SetLineColor(*colour);
+ 		    h->SetMarkerColor(*colour);
+ 		    h->SetMarkerSize(0);
+		    h->SetLineWidth(1.5);
```
Comment (mmusich) [inline Alignment/OfflineValidation/bin/DMRtrends.cc]: @cardinia, this line seems to be the culprit of the clang compilation warning:
```
Alignment/OfflineValidation/bin/DMRtrends.cc
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_6_X_2019-03-18-2300/src/Alignment/OfflineValidation/bin/DMRtrends.cc:598:23: warning: implicit conversion from 'double' to 'Width_t' (aka 'short') changes value from 1.5 to 1 [-Wliteral-conversion]
```
can you fix it?

Comment (cardinia) [reply Alignment/OfflineValidation/bin/DMRtrends.cc]: Yes, it's an easy fix. Unfortunately I got sick, but I'll fix it once I'm back to work.

**Comment by fabiocos** [comment]
please test

**Comment by cardinia** [comment]
A user in our group found a bug in the code I was unaware of. I am currently doing some testing to fix it. Please wait to merge the pull request, I will make a commit soon with the bugfix.

**Comment by pohsun** [comment]
please test

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+1

the size of added and deleted files (clearly a mistake) looks limited, and they are ascii files

File: Alignment/OfflineValidation/python/TkAlAllInOneTool/plottingOptions.py [hunk 2]
```diff
@@ -170,8 +170,6 @@ class PlottingOptionsZMuMu(BasePlottingOptions):
                 "rebineta": "2",
                 "rebinpt": "8",
                 "AutoSetRange": "false",                
-                "CustomMinY": "90.85",
-                "CustomMaxY": "91.5",
```
Comment (adewit) [inline Alignment/OfflineValidation/python/TkAlAllInOneTool/plottingOptions.py]: @cardinia I know this PR is closed, but this change in `TkAlAllInOneTool/plottingOptions.py` actually breaks the Z->mumu validation. Can you submit another PR to revert this change?

Comment (cardinia) [reply Alignment/OfflineValidation/python/TkAlAllInOneTool/plottingOptions.py]: @adewit , those default values were only valid for the Z resonance, and caused a bug when running on the Y resonance.
It's a bug that Rainer found a while ago. There are already default values implemented in the validation tools that depend on the chosen resonance, this change just prevents using the wrong range when running on a different resonance without specifying the range in the config file.

Comment (hroskes) [reply Alignment/OfflineValidation/python/TkAlAllInOneTool/plottingOptions.py]: Hi @cardinia and @adewit, I've been thinking for a while about how to improve the implementation of X->mumu in the all in one tool.  I think the nicer way to go is to make a base class XMuMuValidation and then make subclasses ZMuMuValidation, Y1SMuMuValidation, etc. and similar for plotting options.  Then the defaults can live in the header and be different for each validation.  It's also more correct in the bookkeeping and would let you run multiple resonances->mumu in the same ini file if you wanted.
